File: Algorithms/Golang/Easy/SortArrayByParity.go

/*
Given an integer array nums, move all the even integers
 at the beginning of the array followed by all the odd integers.
Return any array that satisfies this condition.
*/

func sortArrayByParity(nums []int) []int {
	index := 0

	for i := 0; i < len(nums); i++ {
		if nums[i]%2 == 0 {
			nums[index], nums[i] = nums[i], nums[index]
			index++
		}
	}
	return nums
}

File: Algorithms/С++/Easy/SortArrayByParity.cpp

/*
Given an integer array nums, move all the even integers at the beginning 
of the array followed by all the odd integers.
Return any array that satisfies this condition.
*/


class Solution {
public:
	vector<int> sortArrayByParity(vector<int>& nums) {
		for (int i = 0, j = 0; j < nums.size(); j++) {
			if (nums[j] % 2 == 0) {
				swap(nums[i++], nums[j]);
			}
		}
		return nums;
	}
};


// #2

class Solution {
public:
	vector<int> sortArrayByParity(vector<int>& nums) {
		std::partition(nums.begin(), nums.end(), [](int i) {
			return i % 2 == 0;
			});
		return nums;
	}
};